### ISSUE TITLE
Rather than implement a inherent method named default, have Options implement the Default type

### ISSUE THREAD

**Issue by softprops** [mcasper/dogstatsd-rs#14]
this allows for more flexible usage and you get the `Options::default()` method for free

This also allows for more flexible custom construction

```rust
let ops = Options {
  to_addr: "host",
 ..Default::default()
};
```

**Comment by mcasper** [comment]
Definitely, that makes way more sense! If you want to submit a PR that converts us to implementing `Default`, I'd happily accept it

**Comment by mcasper** [comment]
I went ahead and added this in https://github.com/mcasper/dogstatsd-rs/commit/984c96692a8f2244fa0c40218a92e0ad3c8daa93, and it should be available in 0.3.1.

Thanks for the suggestion!
